EIPS/eip-5573.md

```diff
@@ -0,0 +1,273 @@
+
+##### ReCap URI Scheme
+
+A ReCap URI starts with `urn:recap:` followed by `:` and the base64url-encoded payload of the ReCap Details Object. Note, the term base64url is defined in RFC4648 - Base 64 Encoding with URL and Filename Safe Alphabet.
+
+The following is a non-normative example of a ReCap Capability:
+```text
+urn:recap:eyJkZWYiOlsicmVhZCJdLCJ0YXIiOnsibXkucmVzb3VyY2UuMSI6WyJhcHBlbmQiLCJkZWxldGUiXSwibXkucmVzb3VyY2UuMiI6WyJhcHBlbmQiXSwibXkucmVzb3VyY2UuMyI6WyJhcHBlbmQiXX19
+```
+
+##### Ability Strings
+
+Ability Strings identify an action or Ability within a Namespace. They are serialized as `<namespace>/<ability>`. Namespaces and Abilities MUST contain only alphanumeric characters as well as the characters `.`, `*`, `_`, `+`, `-`, conforming to the regex `^[a-zA-Z0-9.*_+-]$`. The ability string as a whole MUST conform to `^[a-zA-Z0-9.*_+-]+\/[a-zA-z0-9.*_+-]+$`. For example, `crud/update` has an ability-namespace of `crud` and an ability-name of `update`.
+
+##### ReCap Details Object Schema
+
+The ReCap Details Object denotes which actions on which resources the delegee is authorized to invoke on behalf of the delegee for the validity period defined in the SIWE message. It can also contain additional information that the RS may require to verify a capability invocation. A ReCap Details Object MUST follow the following JSON Schema:
+
+```jsonc
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "type": "object",
+  "properties": {
+    "att": {
+      "type": "object",
+      "propertyNames": {
+        "format": "uri"
+      },
+      "patternProperties": {
+        "^.+:.*$": {
+          "type": "object",
+          "patternProperties": {
+            "^[a-zA-Z0-9.*_+-]+\/[a-zA-z0-9.*_+-]+$": {
+              "type": "array",
+              "items": {
+                "type": "object"
+              }
+            }
+          },
+          "additionalProperties": false,
+          "minProperties": 1
+        }
+      },
+      "additionalProperties": false,
+      "minProperties": 1
+    },
+    "prf": {
+      "type": "array",
+      "items": {
+        "type": "string",
+        "format": "CID"
+      },
+      "minItems": 1
+    }
+  }
+}
+```
+
+A ReCap Details Object defines the following properties:
+- `att`: (CONDITIONAL) If present, `att` MUST be a JSON object where each key is a URI and each value is an object containing namespaced-actions as keys and a corrosponding value which is an array of qualifications to the action (i.e. a restriction or requirement). The keys of the object MUST be ordered lexicographically.
+- `prf`: (CONDITIONAL) If present, `prf` MUST be a JSON array of string values with at least one entry where each value is a valid Base58-encoded CID which identifies a parent capability, authorising one or more of the entries in `att` if the SIWE `address` does not identify the controller of the `att` entries.
```

Comment:
Make it clear that a parent capability should authorize the Ethereum account that is signing this ReCap. Something like:
```suggestion
- `prf`: (CONDITIONAL) If present, `prf` MUST be a JSON array of string values with at least one entry where each value is a valid Base58-encoded CID which identifies a parent capability, authorising the Ethereum account for one or more of the entries in `att` if the SIWE `address` does not identify the controller of the `att` entries.
```